zokrates_core/src/typed_absy/types.rs

```diff
@@ -156,38 +156,83 @@ impl<'de> Deserialize<'de> for Type {
 
         let mapping = Mapping::deserialize(d)?;
         match mapping.ty.as_str() {
-            "field" => Ok(Type::FieldElement),
-            "bool" => Ok(Type::Boolean),
+            "field" => {
+                if mapping.components.is_some() {
```

Comment:
I feel this would be more idiomatic with a match or something like https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=86d9756a852c677c721add6e6eff77d1 but in this case it's readable

The match seems more readable here

Reply:
replaced the `if` statement with a `match`. 

I've moved the check to a closure to reduce code duplication